### PR TITLE
#24 - cache

### PR DESCRIPTION
**Pull request by giogonzo**

### CODE REVIEW

File: src/AvengerCache.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+const AllowedParams = t.dict(t.Str, AllowedParam, 'AllowedParams');
+
+export function hashedParams(params) {
+  if (process.env.NODE_ENV !== 'production') {
+    AllowedParams(params);
+  }
+  const keys = Object.keys(params);
+  keys.sort((a, b) => a < b ? -1 : a > b ? 1 : 0);
+  const hashed = keys.map(k => `${k}:${params[k]}`).join('-');
+  return hashed ? hashed : '∅';
```
Comment (giogonzo) [inline src/AvengerCache.js]: escape!!111!!!!

File: src/AvengerCache.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+  toJSON() {
+    return this.state;
+  }
+
+  checkParams(params) {
+    return AllowedParams.is(params);
+  }
+
+  get(id, params) {
```
Comment (giogonzo) [inline src/AvengerCache.js]: `id` could be just one more param

**Comment by utaal** [comment]
Ready to merge. @giogonzo please review

**Comment by utaal** [comment]
Rebased. @gcanti you're welcome to take a look as well, the code isn't the prettiest but it seems to work.

**Comment by utaal** [comment]
Not a single comment? Rly?

**Pull request merged**